src/views/SetUnitView/SetUnitView.tsx

```diff
@@ -0,0 +1,217 @@
+                color="secondary"
+                defaultValue={model.name}
+                onChange={handleNameChange}
+              />
+
+              <Grid
+                container
+                direction="column"
+                alignItems="flex-end"
+                justifyContent="flex-end"
+              >
+                <Typography variant="h5" color="primary">
+                  {selectedUnit === null ? "SELECT UNIT" : selectedUnit}
+                </Typography>
+                <Tooltip title="Current Unit Cost">
+                  <Typography color="secondary" variant="subtitle1">
+                    {unitCost} POINTS
+                  </Typography>
+                </Tooltip>
+                <Tooltip title="Remaining Warband Points">
+                  <Typography
+                    sx={(theme) => ({
+                      color:
+                        calculatePointsRemaining < 0
+                          ? theme.palette.error.main
+                          : "text.disabled"
+                    })}
+                    variant="subtitle2"
+                  >
+                    {calculatePointsRemaining} Points Remaining
+                  </Typography>
+                </Tooltip>
+              </Grid>
+            </Grid>
+            <Grid>
+              <SetUnitTabs value={tabValue} handleChange={handleTabChange} />
+              {tabValue === 0 && <SetUnitUnitHeader />}
+              <Grid
+                container
+                maxWidth="6200px"
+                maxHeight="390px"
+                sx={{
+                  overflowX: "hidden",
+                  overflowY: "scroll",
+                  "::-webkit-scrollbar": {
+                    display: "none"
+                  }
+                }}
+              >
+                {(() => {
+                  switch (tabValue) {
+                    case 0:
+                      return (
+                        <>
+                          {unitOptions.map((unit) => (
+                            <SetUnitUnit
+                              unit={unit}
+                              handleClick={() => {
+                                setSelectedUnit(unit.name);
+                                setUnitCost(unit.points);
+                                setModel({ ...model, unit });
+                              }}
+                              isSelected={selectedUnit === unit.name}
+                            />
+                          ))}
+                        </>
+                      );
+                    case 1:
+                      return <Typography>two</Typography>;
+                    case 2:
+                      return <Typography>three</Typography>;
+                    case 3:
+                      return <Typography>four</Typography>;
+
+                    default:
+                      // eslint-disable-next-line no-console
+                      return <p>Unknown tab</p>;
+                  }
```

Comment:
I know you are way ahead of this, but I at some point I think that this should probably be its own component.

All the functions/variables that this section references might need to be in a hook or something, I can help you set this up